optimum/modeling_base.py

```diff
@@ -117,6 +118,8 @@ def save_pretrained(
         os.makedirs(save_directory, exist_ok=True)
 
         self.config.save_pretrained(save_directory)
+        for preprocessor in self._preprocessors:
```

Comment:
`PreTrainedModel` does not save preprocessors with `save_pretrained`, why would we introduce this asymmetry in Optimum? Personnally I don't understand the motivation for this `self._preprocessors`. What's the issue with fully dissociating preprocessing and modeling, like it's done in transformers?

Reply:
If it is not there, we do not do anything, otherwise we save it. The reason for this is because let's say I export a model with the `ORTModel` classes, I want to be able to use the save directory as a model repo end-to-end. Same thing for graph optimization / quantization. Otherwise we would need to provide a different name for the `ORTModel.from_pretrained` and the `AutoTokenizer.from_pretrained`.